src/librustdoc/html/static/main.js

```diff
@@ -385,17 +385,17 @@ function hideThemeButtonState() {
             if (parent && hasClass(parent, "impl-items")) {
                 // In case this is a trait implementation item, we first need to toggle
                 // the "Show hidden undocumented items".
-                onEachLazy(parent.getElementsByClassName("collapsed"), function(e) {
-                    if (e.parentNode === parent) {
+                onEachLazy(parent.childNodes, function(e) {
+                    if (e.tagName === "DETAILS") {
                         // Only click on the toggle we're looking for.
-                        e.click();
+                        e.open = true;
                         return true;
                     }
                 });
                 if (isHidden(elem)) {
                     // The whole parent is collapsed. We need to click on its toggle as well!
-                    if (hasClass(parent.lastElementChild, "collapse-toggle")) {
-                        parent.lastElementChild.click();
+                    if (parent.lastElementChild.tagName === "DETAILS") {
+                        parent.lastElementChild.open = true;
                     }
                 }
             }
```

Comment:
I think this whole section can be replaced by `expandSection(savedHash.slice(1))`;